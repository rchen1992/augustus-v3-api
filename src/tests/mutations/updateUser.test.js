const { testQuery } = require('@tests/utils');
const { mockUsers } = require('@tests/mockData/dataSet');

const userId = '1';
const userName = 'updated username';

testQuery({
    id: 'Update user',
    query: `
        mutation updateUser($fields: UpdateUserInput!) {
            updateUser(fields: $fields) {
                id
                userName
            }
        }
    `,
    variables: {
        fields: {
            userName,
        },
    },
    context: () => {
        return {
            currentUser: {
                user_id: userId,
            },
            services: {
                user: {
                    updateUser() {
                        const user = mockUsers[0];
                        return {
                            ...user,
                            user_name: userName,
                        };
                    },
                },
            },
        };
    },
    expected: undefined,
});
