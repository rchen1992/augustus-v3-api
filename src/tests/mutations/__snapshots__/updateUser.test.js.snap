// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query: Update user 1`] = `
Object {
  "data": Object {
    "updateUser": null,
  },
  "errors": Array [
    [GraphQLError: role "raymondchen" does not exist],
  ],
}
`;
